rainier-core/src/main/scala/com/stripe/rainier/core/Continuous.scala

```diff
@@ -206,3 +217,41 @@ object Uniform {
   def apply(from: Real, to: Real): Continuous =
     standard.scale(to - from).translate(from)
 }
+
+/**
+  * A mixture distribution composed of continuous distributions
+  *
+  * @param pmf A Map with keys representing component beta distributions and values corresponding to the probabilities of those components.
+  * @param _supportTransform A function mapping a real-valued variable to the range of support of the combined mixture.
+  * @param _supportTransformLogJacobian The jacobian (derivative with respect to v) of the _supportTransform function
+  */
+final case class ContinuousMixture(
+    pmf: Map[Continuous, Real],
+    _supportTransform: Variable => Real,
+    _supportTransformLogJacobian: Variable => Real)
+    extends Continuous {
+  def realLogDensity(t: Real): Real =
+    Real
+      .sum(pmf.toList.map {
+        case (dist, prob) =>
+          (dist.realLogDensity(t) + prob.log).exp
+      })
+      .log
+
+  def generator: Generator[Double] =
+    Categorical(pmf).generator.flatMap { d =>
+      d.generator
+    }
+
+  override def supportTransform(v: Variable): Real = _supportTransform(v)
+  override def supportTransformLogJacobian(v: Variable): Real =
+    _supportTransformLogJacobian(v)
+
+}
+
+object ContinuousMixture {
+  def apply(pmfs: Seq[(Continuous, Real)]): ContinuousMixture =
```

Comment:
This API is too implicit for me. I think if we were doing it without a separate `Support` object I'd suggest having the first component be more special. Something like:

```
def apply(first: Continuous, rest: Map[Continuous,Real]): ContinuousMixture = ...
def apply(first: Continuous, second: Continuous, pSecond: Real): ContinuousMixture = apply(first, Map(second -> pSecond))
```

Better though would be to have `support` on `Continuous` and either dynamically check that `support` matches across all of them, or (better/fancier) have a `union` operation on `Support`

Reply:
I like the `Support` object. A `union` operation should be easy enough if `isDefined` is a `Real => Bool` function, which we can `\/` over the constituent parts of the union.

Reply:
Actually I guess it would be a`Real => Real` using `If`

Reply:
Though that is a lot of `If` checks which might cause a slowdown or bad derivatives... wdyt?

Reply:
I think we can just enumerate the cases for a union b, rather than something generic based om isDefinedAt

Reply:
Oh, we'd possibly need a new transform for the new support as well as an `isDefined`. Damn.